service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/IotHubConnectionString.java

```diff
@@ -65,7 +64,9 @@
     // configurations
     private static final String URL_PATH_APPLY_CONTENT_CONFIGURATION = "applyConfigurationContent";
 
-    protected IotHubConnectionString() {
+    protected IotHubConnectionString()
```

Comment:
This file has a big diff, but the net result of my changes are to copy every instance method in here and make a static variant that takes the hostname as the first parameter. Since the service client APIs will no longer be able to build an instance of an IotHubConnectionString, they will need these static overloads

Reply:
Are the static and instance variants both still used? Is the instance variant for the "connection string" flow while the static variant is for the "token credential" flow?

Reply:
I wonder if there is a way to combine these? Can we refer to a singleton instance, instead of calling static methods on this class?

Reply:
The connection string class has a lot of validation baked in, so it isn't easy to instantiate one without an actual connection string. I was tempted to just turn the instance methods into static methods, but this class is potentially used by customers so I'd prefer to avoid removing anything. At this point, internally, we will only use the static methods though

Reply:
Are we passing the IotHubConnectionString class downstream? Wondering if you can do something else instead of this. Something have two different child classes for connection string and token credentials. The downstream should mostly need a way to get a token and that can be implemented in each child class based on the credential type.